### PR TITLE
176091259 chem lab setup

### PR DESCRIPTION
**Pull request by mklewandowski**
This PR adds the infrastructure needed for the chem lab including the test result state and the main components and exports.  None of the test content is in place yet, but users can press the test step buttons to progress through the various tests and have their test result state saved and restored.  Changes include:
- add exports to `chem-utils` that define the interfaces and structures needed to define the chem tests and test results
- add `chemistryTestResults` to the sim output state and function to update this state
- add ChemResutls` component that shows the first page in the chem notebook where the test progress and results are summarized
- add `ChemTest` component that displays a single chem lab test

Can test here:
https://leaf-pack.concord.org/branch/chem-lab-setup/

NOTE: some of the exports in `chem-utils` and the handling of changing the test results will change once we get deeper into implementing the actual tests.

chem-tests

### CODE REVIEW

**Comment by mklewandowski** [comment]
@kswenson I added the chem lab result to the thumbs.  If the user starts any test, then I show the C icon.

File: src/utils/chem-utils.ts [hunk 1]
```diff
@@ -1 +1,88 @@
...
+
+export interface ChemistryTest {
+  type: ChemTestType,
+  label: string,
+  steps: ChemTestStep[],
+  values: ChemTestValue[],
+  units: string,
+}
+
+export const chemistryTests: ChemistryTest[] = [
```
Comment (kswenson) [inline src/utils/chem-utils.ts:49]: I would probably have gone with a map here as well.

**Review by kswenson — APPROVED** [review]
👍 I think this is fine. It seems like the choice of arrays for chemistry tests results in a lot of linear searches (e.g. array.find()) where a map would mean a simple lookup would suffice. I probably would have gone with maps and used for..in for looping, but that may be too much of a change at this point.

File: src/components/notebook/chem-test.tsx [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+  chemistryTest: ChemistryTest;
+  testIndex: number;
+  chemistryTestResults: ChemistryTestResult[];
+  onUpdateTestResult: (type: ChemTestType, completedStep: number) => void;
+}
+
+export const ChemTest: React.FC<IProps> = (props) => {
+  const { chemistryTest, testIndex, chemistryTestResults, onUpdateTestResult } = props;
+  const testResult = chemistryTestResults.find((result) => result.type === chemistryTest.type);
+  const stepsComplete = testResult ? testResult?.stepsComplete : 0;
```
[the remark this answers is not in the document]
Comment (mklewandowski) [reply src/components/notebook/chem-test.tsx]: Fixed

File: src/components/app.tsx [hunk 3]
```diff
@@ -69,6 +71,14 @@ export const App: React.FC<IAppProps<IModelInputState, IModelOutputState, IModel
...
                           showTray: false,
                           trayObjects: [],
+                          chemistryTestResults: [
+                            {type: ChemTestType.airTemperature, stepsComplete: 0, value: 0},
+                            {type: ChemTestType.waterTemperature, stepsComplete: 0, value: 0},
+                            {type: ChemTestType.pH, stepsComplete: 0, value: 0},
+                            {type: ChemTestType.nitrate, stepsComplete: 0, value: 0},
+                            {type: ChemTestType.turbidity, stepsComplete: 0, value: 0},
+                            {type: ChemTestType.dissolvedOxygen, stepsComplete: 0, value: 0}
+                          ],
```
[the remark this answers is not in the document]
Comment (mklewandowski) [reply src/components/app.tsx:81]: I'll leave for now due to time constraints.  I know I also made the switch from array to record in Plant Restoration and ended up regretting it as it introduced other complications.  Not saying that will happen here however, just noting.  Nonetheless, if I free up time once this is further along, I'll circle back and see if a map makes things easier.

**Pull request merged**